src/test/java/com/melon/portfoliomanager/UserControllerTest.java

```diff
@@ -0,0 +1,81 @@
+package com.melon.portfoliomanager;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.melon.portfoliomanager.controllers.UserController;
+import com.melon.portfoliomanager.exceptions.NullFieldException;
+import com.melon.portfoliomanager.exceptions.UsernameAlreadyUsedException;
+import com.melon.portfoliomanager.models.User;
+import com.melon.portfoliomanager.repositories.UserRepository;
+import com.melon.portfoliomanager.services.UserService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.ArrayList;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class UserControllerTest {
+
+    @InjectMocks
+    UserController userController;
+
+    @Mock
+    UserService userService;
+
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    MockMvc mockMvc;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
```

Comment:
Is not the method ```initMocks()``` deprecated ? I think ```openMocks``` is the method that should be used and after you use it you should define @AfterEach() method in which you should run the ```close()``` method.